### PR TITLE
Add defaults from environment

### PR DESCRIPTION
**Pull request by prayerslayer**
So our Docker image becomes more flexible.

### CODE REVIEW

File: costreport/costreport.py [hunk 1]
```diff
@@ -210,32 +209,34 @@ def log_sendmail_note(send_mail):
 
 
 @click.command(context_settings=CONTEXT_SETTINGS)
-@click.argument('entity', default='all')
-@click.argument('serial', default='')
+@click.argument('entity', default=os.getenv("ENTITY"))
```
Comment (tuxlife) [inline costreport/costreport.py]: http://click.pocoo.org/5/options/#values-from-environment-variables

```python
@click.command()
@click.option('--username', envvar='USERNAME')
def greet(username):
    click.echo('Hello %s!' % username)

if __name__ == '__main__':
    greet()
```

File: costreport/costreport.py [hunk 2]
```diff
@@ -210,32 +209,34 @@ def log_sendmail_note(send_mail):
...
+@click.argument('serial', default=os.getenv("SERIAL"))
 @click.option(
```
Comment (tuxlife) [inline costreport/costreport.py]: http://click.pocoo.org/5/options/#choice-options

```python
@click.command()
@click.option('--hash-type', type=click.Choice(['md5', 'sha1']))
def digest(hash_type):
    click.echo(hash_type)
```

File: costreport/costreport.py [hunk 3]
```diff
@@ -405,11 +412,11 @@ def main(
                     send_from,
                     layout_images
                 )
-
                 with open(os.path.join(base_dir, 'costreport.html'), 'w') as f:
                     f.write(html)
-    except:
+    except Exception as e:
         traceback.print_exc()
+        logging.error(e)
```
Comment (tuxlife) [inline costreport/costreport.py]: Why logging **and** tracback.print?

```python
logger.error("Received error: %s", e, exc_info=True)
```

Comment (hjacobs) [inline costreport/costreport.py]: or directly use "logging.exception(...)"

**Pull request merged**